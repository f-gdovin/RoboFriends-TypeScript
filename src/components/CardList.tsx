import * as React from 'react';
import { IRobot } from "../interfaces";
import Card from './Card';

interface ICardListProps {
    robots: IRobot[];
}

const CardList: React.SFC<ICardListProps> = ({ robots }) => {
    return (
        <div>
            {robots.map((robot: IRobot, index: number) => (
                <Card
                    key={`robotId${index}`}
                    id={robot.id}
                    name={robot.name}
                    email={robot.email}
                />)
            )}
        </div>
    )
};

export default CardList;