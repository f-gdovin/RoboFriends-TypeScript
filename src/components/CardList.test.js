import React from 'react';
import CardList from './CardList';
import { shallow } from 'enzyme';

it('should render CardList component', () => {
    const robotsMock = [
        {
            email: 'john@gmail.com',
            id: 1,
            name: 'John'
        },

        {
            email: 'jane@gmail.com',
            id: 2,
            name: 'Jane'
        },

        {
            email: 'jessie@gmail.com',
            id: 3,
            name: 'Jessie'
        }
    ];
    expect(shallow(<CardList robots={robotsMock}/>)).toMatchSnapshot();
});