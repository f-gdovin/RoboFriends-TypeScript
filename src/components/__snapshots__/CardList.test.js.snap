// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "john@gmail.com",
              "id": 1,
              "name": "John",
            },
            Object {
              "email": "jane@gmail.com",
              "id": 2,
              "name": "Jane",
            },
            Object {
              "email": "jessie@gmail.com",
              "id": 3,
              "name": "Jessie",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@gmail.com"
          id={1}
          name="John"
/>,
        <Card
          email="jane@gmail.com"
          id={2}
          name="Jane"
/>,
        <Card
          email="jessie@gmail.com"
          id={3}
          name="Jessie"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "robotId0",
        "nodeType": "function",
        "props": Object {
          "email": "john@gmail.com",
          "id": 1,
          "name": "John",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "robotId1",
        "nodeType": "function",
        "props": Object {
          "email": "jane@gmail.com",
          "id": 2,
          "name": "Jane",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "robotId2",
        "nodeType": "function",
        "props": Object {
          "email": "jessie@gmail.com",
          "id": 3,
          "name": "Jessie",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@gmail.com"
            id={1}
            name="John"
/>,
          <Card
            email="jane@gmail.com"
            id={2}
            name="Jane"
/>,
          <Card
            email="jessie@gmail.com"
            id={3}
            name="Jessie"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "robotId0",
          "nodeType": "function",
          "props": Object {
            "email": "john@gmail.com",
            "id": 1,
            "name": "John",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "robotId1",
          "nodeType": "function",
          "props": Object {
            "email": "jane@gmail.com",
            "id": 2,
            "name": "Jane",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "robotId2",
          "nodeType": "function",
          "props": Object {
            "email": "jessie@gmail.com",
            "id": 3,
            "name": "Jessie",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
